src/main/java/edu/cornell/kfs/tax/dataaccess/impl/TransactionRowPRNCBuilder.java

```diff
@@ -306,39 +309,43 @@ void updateTransactionRowsFromWorkflowDocuments(ResultSet rs, T summary) throws
+                        updatedAttributeValues.putStringAttributeForUpdating(summary.transactionDetailRow.vendorTaxNumber.index, vendorTaxNumber);
+                    }
                     
                     // Do tax-type-specific updates.
-                    doTaxSpecificSecondPassRowSetup(rs, summary);
+                    doTaxSpecificSecondPassRowSetup(rs, summary, updatedAttributeValues);
                     
                     // Update other fields as needed.
                     if (StringUtils.isBlank(documentId)) {
-                        rs.updateString(detailRow.documentNumber.index, CUTaxConstants.DOC_ID_ZERO);
+                        updatedAttributeValues.putStringAttributeForUpdating(summary.transactionDetailRow.documentNumber.index, CUTaxConstants.DOC_ID_ZERO);
                     }
-                    rs.updateString(detailRow.documentTitle.index, (document != null && StringUtils.isNotBlank(document.getTitle()))
-                            ? document.getTitle() : CUTaxConstants.DOC_TITLE_IF_NOT_FOUND);
-                    rs.updateString(detailRow.initiatorNetId.index,
+                    updatedAttributeValues.putStringAttributeForUpdating(summary.transactionDetailRow.documentTitle.index,
+                            (document != null && StringUtils.isNotBlank(document.getTitle())) ? document.getTitle() : CUTaxConstants.DOC_TITLE_IF_NOT_FOUND);
+                    updatedAttributeValues.putStringAttributeForUpdating(summary.transactionDetailRow.initiatorNetId.index,
                             StringUtils.isNotBlank(initiatorPrincipalName) ? initiatorPrincipalName : CUTaxConstants.NETID_IF_NOT_FOUND);
                     if (useDateFinalized) {
-                        rs.updateDate(detailRow.paymentDate.index, dateFinalized);
+                        updatedAttributeValues.putSqlDateAttributeForUpdating(summary.transactionDetailRow.paymentDate.index, dateFinalized);
                     }
                     
                     // Update the transaction row.
-                    rs.updateRow();
+                    LOG.info("TransactionRowPRNCBuilder:: updateTransactionRowsFromWorkflowDocuments: Inserting updated second pass data for "
+                            + "rawTransactionDetailId = {}, documentId = {}", rawTransactionDetailId, documentId);
+                    insertUpdatedTransactionDetail(rs, secondPassTransactionInsertStatement, summary, updatedAttributeValues);
                 } else {
                     // If a Foreign Draft or Wire Transfer that wasn't finalized or was in the wrong reporting period, then delete the row.
-                    rs.deleteRow();
+                    LOG.info("TransactionRowPRNCBuilder:: updateTransactionRowsFromWorkflowDocuments: NO data inserted for "
+                            + "rawTransactionDetailId = {}, documentId = {}", rawTransactionDetailId, documentId);
```

Comment:
Another spot where you may want to update or remove a comment as a result of the no-deletion changes.